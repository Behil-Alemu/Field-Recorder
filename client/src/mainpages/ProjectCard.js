import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import {
	Card,
	CardHeader,
	CardBody,
	CardFooter,
	Heading,
	Button,
	Text,
	IconButton,
	Editable,
	EditableInput,
	EditablePreview,
	Input,
	ButtonGroup,
	Flex,
	useEditableControls
} from '@chakra-ui/react';
import { EditIcon, DeleteIcon, CheckIcon, CloseIcon } from '@chakra-ui/icons';
import { useNavigate } from 'react-router-dom';

import { handleDeleteClick, handleEditClick } from './delete-edit/handleEditDelete.js';
import ConfirmDelete from '../helpers/ConfirmDelete.js';

function ProjectCard({ id, folderName, updateProjects }) {
	let history = useNavigate();
	const [ isOpen, setIsOpen ] = useState(false);
	const [ sampleIdToDelete, setSampleIdToDelete ] = useState(0);
	const onClose = () => setIsOpen(false);

	console.debug('ProjectCard', 'id=', id, 'folderName=', folderName);

	const handleDelete = (id) => {
		setSampleIdToDelete(id);
		setIsOpen(true);
	};

	const handleConfirmDelete = async () => {
		try {
			await handleDeleteClick(sampleIdToDelete);
			updateProjects(sampleIdToDelete);
			history('/homepage');
			return;
		} catch (err) {
			console.error(err);
		}
	};
	const handleEdit = async (id, newValue) => {
		let dataFormat = {
			folderName: newValue
		};

		try {
			let result = await handleEditClick(id, dataFormat);
			console.log(result);
			history('/homepage');
			return;
		} catch (err) {
			console.error(err);
		}
	};
	function EditableControls() {
		const { isEditing, getSubmitButtonProps, getCancelButtonProps, getEditButtonProps } = useEditableControls();

		return isEditing ? (
			<ButtonGroup justifyContent="center" size="sm">
				<IconButton icon={<CheckIcon />} {...getSubmitButtonProps()} />
				<IconButton icon={<CloseIcon />} {...getCancelButtonProps()} />
			</ButtonGroup>
		) : (
			<Flex justifyContent="center">
				<IconButton size="sm" icon={<EditIcon />} {...getEditButtonProps()} />
			</Flex>
		);
	}

	return (
		<div>
			<Card>
				<CardHeader>
					<Heading size="md">
						<Editable
							textAlign="center"
							defaultValue={folderName}
							fontSize="2xl"
							isPreviewFocusable={false}
							onSubmit={(value) => handleEdit(id, value)}
						>
							<EditablePreview />
							<Input as={EditableInput} />
							<EditableControls />
						</Editable>
					</Heading>
				</CardHeader>
				<CardBody>
					<Text>Project # {id}</Text>
					<Link to={`/homepage/${folderName}/${id}`}>
						<Button type="submit" size="sm">
							View samples
						</Button>
					</Link>
				</CardBody>
				<CardFooter>
					<IconButton
						m="2"
						variant="outline"
						aria-label="Search database"
						icon={<DeleteIcon />}
						onClick={() => handleDelete(id)}
					/>
				</CardFooter>
			</Card>
			<ConfirmDelete handleConfirmDelete={handleConfirmDelete} onClose={onClose} isOpen={isOpen} />
		</div>
	);
}

export default ProjectCard;
