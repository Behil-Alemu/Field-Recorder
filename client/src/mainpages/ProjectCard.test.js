import React from 'react';
import { render } from '@testing-library/react';
import ProjectCard from './ProjectCard';
import { MemoryRouter } from 'react-router';

it('matches snapshot without logo', function() {
	const { asFragment } = render(
		<MemoryRouter>
			<ProjectCard folderName="cat" />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
