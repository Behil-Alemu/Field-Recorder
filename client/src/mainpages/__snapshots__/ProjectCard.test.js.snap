// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <div>
    <div
      class="chakra-card css-2dlz7"
    >
      <div
        class="chakra-card__header css-0"
      >
        <h2
          class="chakra-heading css-0"
        >
          <div
            class="chakra-editable css-8nel2v"
          >
            <span
              class="chakra-editable__preview css-10z6p3a"
            >
              cat
            </span>
            <input
              class="chakra-editable__input chakra-input css-f00h3x"
              hidden=""
              value="cat"
            />
            <div
              class="css-1l4w6pd"
            >
              <button
                aria-label="Edit"
                class="chakra-button css-p7mkev"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-onkibi"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g
                    fill="none"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-width="2"
                  >
                    <path
                      d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                    />
                    <path
                      d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                    />
                  </g>
                </svg>
              </button>
            </div>
          </div>
        </h2>
      </div>
      <div
        class="chakra-card__body css-0"
      >
        <p
          class="chakra-text css-0"
        >
          Project # 
        </p>
        <a
          href="/homepage/cat/undefined"
        >
          <button
            class="chakra-button css-4xx2wk"
            type="submit"
          >
            View samples
          </button>
        </a>
      </div>
      <div
        class="chakra-card__footer css-k008qs"
      >
        <button
          aria-label="Search database"
          class="chakra-button css-1xjzpfp"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              fill="currentColor"
            >
              <path
                d="M19.452 7.5H4.547a.5.5 0 00-.5.545l1.287 14.136A2 2 0 007.326 24h9.347a2 2 0 001.992-1.819L19.95 8.045a.5.5 0 00-.129-.382.5.5 0 00-.369-.163zm-9.2 13a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zm5 0a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zM22 4h-4.75a.25.25 0 01-.25-.25V2.5A2.5 2.5 0 0014.5 0h-5A2.5 2.5 0 007 2.5v1.25a.25.25 0 01-.25.25H2a1 1 0 000 2h20a1 1 0 000-2zM9 3.75V2.5a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v1.25a.25.25 0 01-.25.25h-5.5A.25.25 0 019 3.75z"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
